src/Induction/WellFounded.agda

```diff
@@ -112,6 +113,24 @@ module FixPoint
   unfold-wfRec : ∀ {x} → wfRec P f x ≡ f x (λ y _ → wfRec P f y)
   unfold-wfRec {x} = f-ext x wfRecBuilder-wfRec
 
+------------------------------------------------------------------------
+-- Well-founded relations are asymmetric and irreflexive.
+
+module _ {_<_ : Rel A r} where
+  Acc-asymm : (x : A) → (Acc _<_ x) → (y : A) → x < y → ¬ (y < x)
+  Acc-asymm = Some.wfRec _ λ x hx y x<y y<x → hx y y<x x y<x x<y
```

Comment:
Firstly: naming
I think that the ambient style of the library calls `Asymmetric` proofs `asym`, so I'd probably call this `Acc-asym`?

Secondly: implicit/explicit quantification
At the cost of some clunkiness on the RHS of this definition, I would be tempted to refactor the type of `Acc-asym(m)` to be
```agda
∀ {x y} → (Acc _<_ x) → x < y → ¬ (y < x)
```

Reply:
`acc-asym` is good. Also don't need brackets around `Acc _<_ x`.